libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -191,86 +192,150 @@ func setupIPChains(config configuration, version iptables.IPVersion) (retErr err
 	return nil
 }
 
-func (n *bridgeNetwork) setupIP4Tables(config *networkConfiguration, i *bridgeInterface) error {
-	maskedAddrv4 := &net.IPNet{
-		IP:   i.bridgeIPv4.IP.Mask(i.bridgeIPv4.Mask),
-		Mask: i.bridgeIPv4.Mask,
-	}
-	return n.setupIPTables(iptables.IPv4, maskedAddrv4, config, i)
+type networkConfigFam struct {
+	HostIP      netip.Addr
+	Prefix      netip.Prefix
+	Routed      bool
+	Unprotected bool
+}
+
+type networkConfig struct {
+	IfName     string
+	Internal   bool
+	ICC        bool
+	Masquerade bool
+	Config4    networkConfigFam
+	Config6    networkConfigFam
+	Hairpin    bool
+	Enable4    bool
+	Enable6    bool
+}
+
+type iptablesNetwork struct {
+	networkConfig
+	cleanFuncs iptablesCleanFuncs
```

Comment:
I think clean up functions aren't even needed if all delete functions are made idempotent. That is, if `configure` fails, it's caller's responsibility to delete the network altogether. This would also solve the zombie rules issue described here: https://github.com/moby/moby/pull/49668/files#diff-93cf2916a163f32710fd2d4cc0845eea796989ba93c0dc2c9418f3eaeab4adccR10-R11

Maybe something to keep in mind for a follow-up?